src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs

```diff
@@ -0,0 +1,232 @@
+        {
+            var textDocumentPositionParams = input.ToObject<TextDocumentPositionParams>();
+            return await this._protocol.GetDocumentHighlightAsync(_workspace.CurrentSolution, textDocumentPositionParams, _clientCapabilities, cancellationToken).ConfigureAwait(false);
+        }
+
+        [JsonRpcMethod(Methods.TextDocumentDocumentSymbolName)]
+        public async Task<object[]> GetTextDocumentDocumentSymbolsAsync(JToken input, CancellationToken cancellationToken)
+        {
+            var documentSymbolParams = input.ToObject<DocumentSymbolParams>();
+            return await this._protocol.GetDocumentSymbolsAsync(_workspace.CurrentSolution, documentSymbolParams, _clientCapabilities, cancellationToken).ConfigureAwait(false);
+        }
+
+        [JsonRpcMethod(Methods.TextDocumentFormattingName)]
+        public async Task<TextEdit[]> GetTextDocumentFormattingAsync(JToken input, CancellationToken cancellationToken)
+        {
+            var documentFormattingParams = input.ToObject<DocumentFormattingParams>();
+            return await this._protocol.FormatDocumentAsync(_workspace.CurrentSolution, documentFormattingParams, _clientCapabilities, cancellationToken).ConfigureAwait(false);
+        }
+
+        [JsonRpcMethod(Methods.TextDocumentOnTypeFormattingName)]
+        public async Task<TextEdit[]> GetTextDocumentFormattingOnTypeAsync(JToken input, CancellationToken cancellationToken)
+        {
+            var documentOnTypeFormattingParams = input.ToObject<DocumentOnTypeFormattingParams>();
+            return await this._protocol.FormatDocumentOnTypeAsync(_workspace.CurrentSolution, documentOnTypeFormattingParams, _clientCapabilities, cancellationToken).ConfigureAwait(false);
+        }
+
+        [JsonRpcMethod(Methods.TextDocumentImplementationName)]
+        public async Task<object> GetTextDocumentImplementationsAsync(JToken input, CancellationToken cancellationToken)
+        {
+            var textDocumentPositionParams = input.ToObject<TextDocumentPositionParams>();
+            return await this._protocol.FindImplementationsAsync(_workspace.CurrentSolution, textDocumentPositionParams, _clientCapabilities, cancellationToken).ConfigureAwait(false);
+        }
+
+        [JsonRpcMethod(Methods.TextDocumentRangeFormattingName)]
+        public async Task<TextEdit[]> GetTextDocumentRangeFormattingAsync(JToken input, CancellationToken cancellationToken)
+        {
+            var documentRangeFormattingParams = input.ToObject<DocumentRangeFormattingParams>();
+            return await this._protocol.FormatDocumentRangeAsync(_workspace.CurrentSolution, documentRangeFormattingParams, _clientCapabilities, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// FAR calls to get third party references (xaml) and to get those references it expectes to be called on the UI thread.
```

Comment:
Ditto as earlier: move the comment out of the xml doc comment tag and by the SwitchToMainThread (wherever it ends up)